### ISSUE TITLE
[bug] nchan_message_timeout deletes channel

### ISSUE THREAD

**Issue by bymaximus** [slact/nchan#234]
Seems when using ```nchan_message_timeout``` with redis storage nchan sets channel TTL after posting fritst message which makes entire channel get removed from redis making nchan unable to know if any subscriber (websocket) is still connected to channel.

How to reproduce:
1. make a GET call to post endpoint, it will show 404
2. start websocket subscriber connection, it will create channel
3. make step 1 GET again, it will show channel status, 1 subscriber, 0 messages
4. make a POST call to post endpoint, it will show POST result
5. wait ```nchan_message_timeout``` time and then make step 1 GET again, it will show 404 while still subscriber connected, should show channel status (not expire channel, or expire channel using ```nchan_redis_idle_channel_cache_timeout```

```
        nchan_redis_url redis://XXXX/0;
        nchan_use_redis on;
        nchan_subscriber_first_message newest;
        nchan_store_messages off;
        nchan_message_timeout 5s;
        #nchan_redis_idle_channel_cache_timeout 5s;
        nchan_redis_idle_channel_cache_timeout 0s;
        nchan_redis_ping_interval 30s;
        #nchan_channel_timeout 10s;

        location = /condition {
                if ($request_method = 'GET') {
                        nchan_publisher http;
                        nchan_channel_id "$arg_channel";
                }
        }
        location = /post {
                if ($request_method = 'GET') {
                        return 405;
                }
                eval_subrequest_in_memory off;
                eval_buffer_size 4k;
                eval_override_content_type text/plain;
                eval $channelStatus {
                        ..........
                        #get here channel status with GET and decide if will POST or not
                        proxy_pass "http://localhost/condition?channel=$arg_channel";
                }
                if ($channelStatus = '') {
                        return 304;
                }
                if ($channelStatus ~ '"subscribers": 0,') {
                        return 204;
                }
                if ($channelStatus ~ '"subscribers": ([1-9]+),') {
                        nchan_publisher http;
                        nchan_channel_id "$arg_channel";
                }
        }
        location ~ /postedData/(\w+)$ {
                nchan_channel_id "$1";
                nchan_subscriber websocket;
        }
```

```
1471975124.084092 [0 XXXXX:48033] "EVALSHA" "43b51538c85d2d2c7950841680553f1bf7dfafa9" "0" "/796qj" "1471975123" "asd" "application/x-www-form-urlencoded" "" "5" "0"
1471975124.084145 [0 lua] "echo" " #######  PUBLISH   ######## "
1471975124.084164 [0 lua] "EXISTS" "{channel:/796qj}"
1471975124.084172 [0 lua] "HGETALL" "{channel:/796qj}"
1471975124.084203 [0 lua] "EXISTS" "{channel:/796qj}:msg:1471975123:0"
1471975124.084214 [0 lua] "HSET" "{channel:/796qj}" "current_message" "1471975123:0"
1471975124.084238 [0 lua] "HSET" "{channel:/796qj}" "time" "1471975123"
1471975124.084252 [0 lua] "HSET" "{channel:/796qj}" "ttl" "5"
1471975124.084263 [0 lua] "HSET" "{channel:/796qj}" "max_stored_messages" "0"
1471975124.084298 [0 lua] "HMSET" "{channel:/796qj}:msg:1471975123:0" "unbuffered" "1" "content_type" "application/x-www-form-urlencoded" "id" "1471975123:0" "data" "asd" "prev_tag" "0" "time" "1471975123" "eventsource_event" "" "prev_time" "0" "ttl" "5" "tag" "0"
1471975124.084345 [0 lua] "EXPIRE" "{channel:/796qj}:msg:1471975123:0" "5"
HERE ========> 1471975124.084358 [0 lua] "EXPIRE" "{channel:/796qj}" "5" <========HERE
1471975124.084367 [0 lua] "EXPIRE" "{channel:/796qj}:messages" "5"
1471975124.084376 [0 lua] "EXPIRE" "{channel:/796qj}:subscribers" "5"
1471975124.084435 [0 lua] "PUBSUB" "NUMSUB" "{channel:/796qj}:pubsub"
1471975124.084457 [0 lua] "PUBLISH" "{channel:/796qj}:pubsub" "\x9a\xacmax_msgs+msg\x00\x05\xceW\xbc\x8e\xd3\x00\x00\x00\xa3asd\xd9!application/x-www-form-urlencoded\xa0"
1471975124.084485 [0 lua] "llen" "{channel:/796qj}:messages"
1471975124.253918 [0 XXXX] "EVALSHA" "d995d33aa9707b6e63d2caf68154cb34ce0aa423" "0" "/796qj" "1471975123" "0" "FILO"
1471975124.253987 [0 lua] "echo" " #######  GET_MESSAGE ######## "
```

**Comment by slact** [comment]
This issue is actually a little trickier, because the TTL of the channel keys is supposed to be extended by a keepalive timer as long as there are subscribers. Will investigate.

**Comment by bymaximus** [comment]
This keepalive timer is not ```nchan_redis_ping_interval```, right? Because my ```nchan_redis_ping_interval``` is 30 seconds

**Comment by slact** [comment]
No, this is an adaptive timer that's fired periodically for each channel. ( https://github.com/slact/nchan/blob/master/src/store/redis/rdsstore.c#L1470-L1514 )

**Comment by bymaximus** [comment]
FYI: When testing, I was running redis MONITOR to see redis operations, I didnt saw this '####### CHANNEL KEEPALIVE #######' lua echo like others lua echo's I posted. Seems not being executed

**Comment by slact** [comment]
Yep. I already have a fix in mind, I'm just busy with something else at the moment. I'll have it ready tonight.

**Comment by bymaximus** [comment]
Ok, no problem, was trying to help you with some information about this scenario I got here, awesome you find it.

Big thanks

**Comment by slact** [comment]
fixed in  e24b796

**Comment by bymaximus** [comment]
Awesome, seems working as expected. Thank you!!!
